python/spark_sklearn/random_search.py

```diff
@@ -0,0 +1,246 @@
+"""
+Class for parallelizing RandomizedSearchCV jobs in scikit-learn
+"""
+
+from sklearn.model_selection import ParameterSampler
+from spark_sklearn.base_search import SparkBaseSearchCV
+
+
+class RandomizedSearchCV(SparkBaseSearchCV):
+    """Randomized search on hyper parameters.
+
+    RandomizedSearchCV implements a "fit" and a "score" method.
+    It also implements "predict", "predict_proba", "decision_function",
+    "transform" and "inverse_transform" if they are implemented in the
+    estimator used.
+
+    The parameters of the estimator used to apply these methods are optimized
+    by cross-validated search over parameter settings.
+
+    In contrast to GridSearchCV, not all parameter values are tried out, but
+    rather a fixed number of parameter settings is sampled from the specified
+    distributions. The number of parameter settings that are tried is
+    given by n_iter.
+
+    If all parameters are presented as a list,
+    sampling without replacement is performed. If at least one parameter
+    is given as a distribution, sampling with replacement is used.
+    It is highly recommended to use continuous distributions for continuous
+    parameters.
```

Comment:
Can you comment on why? or is it just documentation copied from sklearn?

Reply:
It's copied from scikit. I suppose it's just that it makes sense to pick a distribution with continuous support for continuous params or else only integer values will be tried, and that doesn't sound that sensible. Not sure if I have a great extra comment here.